From f2c62e0c0fc9dcd487e87db49fc5cc818f5ba97b Mon Sep 17 00:00:00 2001
From: Zach Reizner <zachr@google.com>
Date: Fri, 20 Oct 2017 13:42:22 -0700
Subject: [PATCH] Update writeArchive handnling for std::error_code

The writeArchive function was changed to return a std::error_code instead of a
pair. Latest LLVM actually uses a new llvm::Error type, but this change is
designed to deal with LLVM checkouts slightly older than r313937.
---
 src/rustllvm/ArchiveWrapper.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/rustllvm/ArchiveWrapper.cpp b/src/rustllvm/ArchiveWrapper.cpp
index 7f76861c07..61109b802a 100644
--- a/src/rustllvm/ArchiveWrapper.cpp
+++ b/src/rustllvm/ArchiveWrapper.cpp
@@ -285,8 +285,14 @@ LLVMRustWriteArchive(char *Dst, size_t NumMembers,
 #else
   auto Pair = writeArchive(Dst, Members, WriteSymbtab, Kind, true);
 #endif
+#if LLVM_VERSION_GE(5, 0)
+  if (!Pair)
+     return LLVMRustResult::Success;
+  LLVMRustSetLastError(Pair.message().c_str());
+#else
   if (!Pair.second)
     return LLVMRustResult::Success;
   LLVMRustSetLastError(Pair.second.message().c_str());
+#endif
   return LLVMRustResult::Failure;
 }
-- 
2.15.0.rc0.271.g36b669edcc-goog
